diagnostic.md

```diff
@@ -2,37 +2,51 @@
 
 What are three advantages of using Ember?
 
-```sh
-# your answer here
+```md
+A routing system that allows for a single page application allowing for
+  back functionality and boomarking
+Less AJAX
+More use of the cache because of less AJAX, yielding more responsiveness
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
-```sh
-# your answer here
+```md
+Each template is written in Handlebars. The template is the HTML content that
+corresponds to the route.
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
-```sh
-# your answer here
+```md
+Routes
+Templates
+Models
 ```
 
 How does a component differ from an Ember template?
 
-```sh
-# your answer here
+```md
+Still kind of confused about this. Components come from the template and don't
+have full access to the route. They are smaller pieces of code and thus, more
+modular.
 ```
 
 What is an opinionated framework? Is Ember opinionated? How or how not?
 
-```sh
-# your answer here
+```md
+An opinionated framework is one in which the person using the framework
+*has* to use the guidelines setup by the creator of the framework. Ember is
+opinionanted because it relies heavily on its own conventions. If using Ember
+and straying from conventions, it could result in someone not being able to read
+your code or Ember not working entirely.
 ```
 
 In Ember what is a route's job (please list three)?
 
-```sh
-# your answer here
+```md
+Takes you to the desired view state
```

Comment:
I would put it this way:
1. Parse information from the url
2. Link the route to a component or template
3. Load the template/UI data via the 'model' hook.

.5/1